### ISSUE TITLE
Fields of Vertex struct being reordered on new rust version results in incorrect render result

### ISSUE THREAD

**Issue by nbvdkamp** [phaazon/luminance-rs#616]
After updating my nightly rust install my luminance application started rendering my models completely incorrectly, as if the attribute data is being garbled. On the 2022-11-11 nightly build it still works  as expected so I'm quite sure https://github.com/rust-lang/rust/pull/102750 being merged is what caused things to break, though I haven't bisected yet.

This is my vertex struct:
```rust
#[derive(Copy, Clone, Debug, Semantics)]
pub enum VertexSemantics {
    #[sem(name = "position", repr = "[f32; 3]", wrapper = "VertexPosition")]
    Position,
    #[sem(name = "normal", repr = "[f32; 3]", wrapper = "VertexNormal")]
    Normal,
    #[sem(name = "uv", repr = "[f32; 2]", wrapper = "VertexUV")]
    TextureCoords,
}

#[derive(Copy, Clone, Vertex)]
#[vertex(sem = "VertexSemantics")]
pub struct Vertex {
    pub position: VertexPosition,
    pub normal: VertexNormal,
    pub uv: VertexUV,
}
```
Adding `#[repr(C)]` to `Vertex` fixes the issue so presumably luminance makes a bad assumption somewhere about vertex struct layout being not being reordered.

**Comment by phaazon** [comment]
Yes, they must be `#[repr(C)]`. I’ll add something to check that in the proc-macro.

**Comment by ColonelThirtyTwo** [comment]
Luminance should use [`memoffset::offset_of`](https://docs.rs/memoffset/latest/memoffset/macro.offset_of.html), [`field_offset::offset_of`](https://docs.rs/field-offset/latest/field_offset/macro.offset_of.html), or equivalent to find the offsets in a way that works with `#[repr(Rust)]`.
